packages/@webex/contact-center/src/services/task/Task.ts

```diff
@@ -17,21 +18,309 @@ import routingContact from './contact';
+
+    return Promise.reject(new Error('endConsult not supported for this channel type'));
+  }
+
+  public async consultTransfer(): Promise<TaskResponse> {
+    this.unsupportedMethodError('consultTransfer');
+
+    return Promise.reject(new Error('consultTransfer not supported for this channel type'));
+  }
+
+  public async consultConference(): Promise<TaskResponse> {
+    this.unsupportedMethodError('consultConference');
+
+    return Promise.reject(new Error('consultConference not supported for this channel type'));
+  }
+
+  public async exitConference(): Promise<TaskResponse> {
+    this.unsupportedMethodError('exitConference');
+
+    return Promise.reject(new Error('exitConference not supported for this channel type'));
+  }
+
+  public async transferConference(): Promise<TaskResponse> {
+    this.unsupportedMethodError('transferConference');
+
+    return Promise.reject(new Error('transferConference not supported for this channel type'));
+  }
+
+  public async toggleMute(): Promise<void> {
+    this.unsupportedMethodError('toggleMute');
+
+    return Promise.reject(new Error('toggleMute not supported for this channel type'));
+  }
+
+  public unregisterWebCallListeners(): void {
+    // Default implementation - child classes can override
+    LoggerProxy.log('unregisterWebCallListeners called', {
+      module: CC_FILE,
+      method: 'unregisterWebCallListeners',
+    });
+  }
+
+  /**
+   * Cancel any in-progress auto wrap-up timer.
+   * Base implementation just clears the timer reference so subclasses inherit the behavior.
+   */
+  public cancelAutoWrapupTimer(): void {
+    if (this.autoWrapup) {
+      this.autoWrapup.clear();
+      this.autoWrapup = undefined;
+      LoggerProxy.log('Auto wrap-up timer cancelled', {
+        module: CC_FILE,
+        method: 'cancelAutoWrapupTimer',
+        interactionId: this.data?.interactionId,
+      });
+    }
+  }
+
+  // Voice tasks use holdResume(), but provide separate methods for interface compliance
+  public async hold(): Promise<TaskResponse> {
+    this.unsupportedMethodError('hold');
+
+    return Promise.reject(new Error('hold not supported for this channel type'));
+  }
+
+  public async resume(): Promise<TaskResponse> {
+    this.unsupportedMethodError('resume');
+
+    return Promise.reject(new Error('resume not supported for this channel type'));
+  }
+
+  public async holdResume(): Promise<TaskResponse> {
+    this.unsupportedMethodError('holdResume');
+
+    return Promise.reject(new Error('holdResume not supported for this channel type'));
+  }
+
+  /**
+   * Get UI controls for task actions.
+   * Computed from state machine state and context.
+   *
+   * @example
+   * ```typescript
+   * const visible = task.taskUiControls.hold.visible;
+   * const enabled = task.taskUiControls.hold.enabled;
+   * ```
+   */
+  public get taskUiControls(): TaskUIActions {
+    // Convert computed UI controls to TaskActionControl objects for backward compatibility
```

Comment:
remove this , make sure we publish event on each UI control change , its will have UI control object